src/runtime/methodbinder.cs

```diff
@@ -201,6 +208,11 @@ internal static int ArgPrecedence(Type t)
                 return 3000;
             }
 
+            if (t.IsAssignableFrom(typeof(PyObject)))
+            {
+                return -1;
```

Comment:
That makes `TypeCode.Object` entry in the `switch` below unreachable. Is it by design?
Also, negative precedence does not sound good. What is its purpose?